### PR TITLE
rev_Update hierarchy of Projects, Project Items...

### PR DESCRIPTION
**Pull request by mlabuda**
Have proper hierarchy - interface, abstract class, implementation.

### CODE REVIEW

**Comment by mlabuda** [comment]
PR job on windows and Mac passed. Rerun job on Ubuntu, there were some failures but probably bcs. of env.

File: plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/condition/WidgetIsFound.java [hunk 1]
```diff
@@ -85,8 +89,22 @@
 	public boolean test() {
 		
 		properWidget = widgetLookup.activeWidget(parent, am, index);
-
-		if(properWidget == null){
+
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/condition/WidgetIsFound.java]: can you please remove this part ?

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/Resource.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
+package org.jboss.reddeer.eclipse.core.resources;
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/Resource.java]: add copyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/ProjectItem.java [hunk 3]
```diff
@@ -1,203 +1,39 @@
-/*******************************************************************************
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/ProjectItem.java]: add compyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/Project.java [hunk 4]
```diff
@@ -1,40 +1,42 @@
-/*******************************************************************************
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/Project.java]: copyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/MavenProject.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package org.jboss.reddeer.eclipse.core.resources;
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/MavenProject.java]: copyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/DefaultResource.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+package org.jboss.reddeer.eclipse.core.resources;
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/DefaultResource.java]: copyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractResource.java [hunk 7]
```diff
@@ -0,0 +1,232 @@
+package org.jboss.reddeer.eclipse.core.resources;
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractResource.java]: copyright

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractProjectItem.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+	public AbstractProjectItem(TreeItem treeItem) {
+		super(treeItem);
+	}
+
+	@Override
+	public void openWith(String editor) {
+		select();
+		new ContextMenu("Open With", "Other...").select();
+		;
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractProjectItem.java]: remove semicolon

File: plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractProjectItem.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+		new WaitUntil(new ShellWithTextIsAvailable("Editor Selection"));
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/core/resources/AbstractProjectItem.java]: you can use new DefaultShell("Editor Selection") and then replace new WaitWhile(new ShellWithTextIsAvailable("Editor Selection")); on line 42 with WaitWhile(ShellIsAvailable(defaultShell))

**Pull request merged**